### PR TITLE
5730 Added support for mbtile files

### PR DESCRIPTION
**Pull request by kellykelly3**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+                        byte[] bimg = connector.getTileBytes(uri.toString());
+                        if (bimg != null && bimg.length > 0) {
+                            img = ImageIO.read(new ByteArrayInputStream(bimg));
+                            cache.put(uri, bimg, img);
+                            img = cache.get(uri);
+                        }
+                    }
+                    if (img == null) {
+                        logger.log(Level.INFO, String.format("Failed to load: %s", uri));
+                    } else {
```
Comment (markmckinnon) [inline Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java]: Maybe add a comment here about what might cause this so when they appear someone can have an idea why it happened.  IE: drilled down to far into map.

File: Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+                    URI uri = getURI(tile);
+                    BufferedImage img = cache.get(uri);
+                    if (img == null) {
+                        img = getImage(uri);
+                    }
+                    if (img != null) {
+                        addImageToTile(tile, img);
+                    }
+                } catch (OutOfMemoryError memErr) {
+                    cache.needMoreMemory();
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java]: Please log the out of memory exception as a WARNING.

File: Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+                } catch (Exception ex) {
+                    if (remainingAttempts == 0) {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/geolocation/MBTilesTileFactory.java]: Let's comment the exception firewall as the work of the original author and leave it in place despite what Codacy has to say.

File: Core/src/org/sleuthkit/autopsy/geolocation/MBTilesFileConnector.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+            // them.
+            int osmZoom = getTotalMapZoom() - zoom;
+            return String.format("%d/%d/%d", osmZoom, x, y);
+        }
+    }
+
+    /**
+     * A ConnectionPool to manage the connections to the mbtile\sql file.
+     */
+    private final class ConnectionPool {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/geolocation/MBTilesFileConnector.java]: Let's take out the connection pool for now.

**Comment by kellykelly3** [comment]
@rcordovano I made the changes to this pull request that you asked for.

**Pull request merged**